### DISCUSSION TITLE
GTA V’s FPS is significantly reduced when running on CXWine

### DISCUSSION THREAD

**Discussion by lockieluke** [The-Wineskin-Project/WineskinServer#341] [General]
FPS is really low when GTAV is running on CXWine instead of regular CrossOver I've installed all the recommended dependencies including DXVK.  I would really like to ditch CrossOver as I cannot financially afford it right now.

**Comment by palmerj** [comment]
Question: how did you get GTA5 running with the latest version of Wineskin? I'm running under Apple silicon? I get it to get the title loading screen but then it locks. I'm using Wineskin-2.9.1.8 engine 22.1.1-rc1. What tricks did you install? Works ok on crossover 22.1.1. Note that on both CXWine (up until lock) and crossover I'm only seeing 30-31 FPS and I'm on a M2Max 16" macbook.

**Reply by Gcenx** [reply]
GTAV requires microphone access or the game won’t run.

Or does it fail before the game launches as nobody had reported any issue for GTAV with any of the current or prior WineCX22.x builds.

**Reply by palmerj** [reply]
On crossover it does a screen calibration and asks for access to my mic on first start. But even after deleting the settings.xml file, on wineskin it does access for access to the mic or do the screen calibration.

**Comment by Gcenx** [comment]
GTAV runs best with dxvk_master last I’d tested this title.

**Reply by palmerj** [reply]
I've created a clean wineskin wrapper with Wineskin-2.9.1.8/WS11WineCX64Bit22.1.1-rc1 and only installed the dxvk_master trick, then copied the pre-installed GTA5 folder and run GTA5.exe. Still doesn't get past the story loading screen. Also on that screen it's loading loading all of the graphic. 

<img width="868" alt="Screenshot 2023-04-29 at 7 58 24 am" src="https://user-images.githubusercontent.com/580022/235262176-90a9e861-c4c1-40ea-80f3-4c65710e6ef4.png">

I do see about 300k lines of `01cc:fixme:keyboard:NtUserActivateKeyboardLayout flags 40000000 not supported`. Full log is [here](https://gist.github.com/palmerj/d091c8fae08d8e8c7bf0648d3cddbf69)

On Crossover 22.1.1 I create an empty bottle with nothing installed, running default setting, and then using the same pre-installed GTA5 folder and run GTA5.exe, everything works well as far as I've tested. 

<img width="868" alt="Screenshot 2023-04-29 at 7 03 16 am" src="https://user-images.githubusercontent.com/580022/235262226-9cbd61cc-68b7-4a84-b103-3914335ad350.png">

I'm on a Macbook pro 16" M2Max 32GB ram. Maybe some issue with the M2Max? One difference I do see is the difference in DXVK driver and vulkan versions. How might I get those same versions with Wineskin? What does driver version 0.2.1918/0.2.2010 refer to?

**Reply by palmerj** [reply]
Interesting comment here about the story loading bug: https://youtu.be/mDrFKlOGH94?t=140. In CX the the DXVK 11 backend needs to be enabled, which is the default. But wineskin is doing that after dxvk_master is installed right?

**Reply by Gcenx** [reply]
Thank you for providing a log I’ll take a closer look once I’m in front of my mac.

I have an idea what might have caused this but would need to confirm, I’d left Steam verifying my own copy of GTAV before heading to work today.

**Reply by Gcenx** [reply]
> Interesting comment here about the story loading bug: https://youtu.be/mDrFKlOGH94?t=140. In CX the the DXVK 11 backend needs to be enabled, which is the default. But wineskin is doing that after dxvk_master is installed right?

Installing any of the DXVK verbs has the same affect as toggling DXVK within the CrossOver UI, you can see it’s enabled and being used from the provided log.

**Reply by palmerj** [reply]
I've just noticed that I only get this issue when I run the software from the wineskin "Run Test" function. I got so busy trying to test the wrapper that I forgot to try just double click the wrapper. Do you know why the run test method is not working?

**Reply by Gcenx** [reply]
> I've just noticed that I only get this issue when I run the software from the wineskin "Run Test" function. I got so busy trying to test the wrapper that I forgot to try just double click the wrapper. Do you know why the run test method is not working?

That’s interesting I’ll need to look into that, though if you enable “Always make Log file, not only when doing test run” will work.

I’d only just figured out _why_ “Rockstar Games Launcher” wasn’t working so resolved that issue.

**Comment by palmerj** [comment]
>  though if you enable “Always make Log file, not only when doing test run” will work.

Yup confirmed that does work. Though with all of the `01d0:fixme:keyboard:NtUserActivateKeyboardLayout flags 40000000 not supported` message my log gets large fast!

**Reply by Gcenx** [reply]
If you look at the WINEDEBUG field you could force expand it to disable any output from `keyboard` like it currently ignores `plugplay` and `esync` related messages.

**Reply by palmerj** [reply]
Ok thank you I will try that

**Comment by palmerj** [comment]
More testing with testing with GTAV with engine 22.1.1-rc1 and DXVK-master I've noticed the FPS is about 30 less (50 vs 85 on M2Max with normal settings with 1440p) than CX 22 and there are missing textures/models. Any idea?

See missing guns in below screenshot

![Screenshot 2023-05-07 at 3 54 52 pm](https://user-images.githubusercontent.com/580022/236660433-2ce61674-ddb5-4952-8511-27d54a5b8529.png)

**Reply by Gcenx** [reply]
Hum try upgrading the wrapper/engine and installing dxvk_master again to get the latest version.

The guns were visible and fps was hovering around 30-40 ish FPS on my base m1 mini so I'd expect better performance with an M2Max.

dxvk_master has a upstream fix https://github.com/Gcenx/DXVK-macOS/commit/e46d9c53264147c5588f0d1668b50b55acc40bfe this might affect FPS but should avoid massive performance drops in desert area.

**Reply by palmerj** [reply]
Does this require me running with DX11 in GTAV? I've been using the DX10 renderer.

**Reply by Gcenx** [reply]
I’d assume I’m using DX11 render as I’ve not changed it

**Reply by palmerj** [reply]
upgrading the wrapper/engine and installing dxvk_master again on DX10 the gun models are back and fixed - thanks. The FPS is still 30FPS lower than CX. Any ideas why that is? Is your dxvk code different to their code?

**Reply by palmerj** [reply]
I tried DX11 with same 1440p and normal settings and still only got 35-40FPS. CX 22 is way faster.

**Reply by Gcenx** [reply]
> The FPS is still 30FPS lower than CX. Any ideas why that is? Is your dxvk code different to their code?

dxvk_master is very different than CrossOver-21+ shipped DXVK it also includes the following fix https://github.com/doitsujin/dxvk/issues/3311 that light affects performance while still generating shader cache. You could try `dxvk1103` this doesn't contain all the extra changes but includes the macOS fixes from CX (all DXVK-macOS archives have macOS fixes builtin) 

If you want the exact same version of DXVK as CrossOver install `dxvk1070` it won't display the same version string but the exact same sources were used.

DXVK-1.7 shader cache and DXVK-1.10.3 shader cache won't be compatible so each time you swap it will get trashed and regenerated.

It could be in CrossOver you've enabled ESYNC where on Wineskin this needs to be enabled manually (currently)

In the "Unix Commands" field add the following
```
export WINEESYNC=1
```

**Reply by palmerj** [reply]
Thanks for the help once again :-)

I tried `export WINEESYNC=1` but that made no difference. I then tried `dxvk1103` and that also made no difference. But when I tried `dxvk1070` I got a much improved FPS. In fact between 100-150!

![Screenshot 2023-05-09 at 5 26 07 pm](https://user-images.githubusercontent.com/580022/237026532-c664c0ed-bbad-40be-b817-dbf1b95aca4f.png)

![Screenshot 2023-05-09 at 5 23 06 pm](https://user-images.githubusercontent.com/580022/237026686-525f48e9-0158-4651-b44e-cf27d17f611c.png)

Any idea why we now have the performance regression?

**Reply by Gcenx** [reply]
Just tried this on my m1 mini and DXVK-1.7 the FPS is less fluid while it does display slightly higher FPS in the HUD at times it feels less smooth and jerky, this leads me to assume you already have a decently generated shader cache for DXVK-v1.7 so it's running better on your system using that verson.

What you could do is manually unpack DXVK-v1.7 x64 dlls into `steamapps/common/Grand Theft Auto V` so GTAV will always use that version while the rest of the wrapper can use `dxvk_master`

**Reply by palmerj** [reply]
Ok cool. After caching I'm definitely getting a higher and more constant FPS experience. But I will have to check out when you are in desert and you look at the direction of the city as per https://github.com/doitsujin/dxvk/issues/3311
